### ISSUE TITLE
Hallucination / Text Addition

### ISSUE THREAD

**Issue by Tejaswgupta** [Vaibhavs10/insanely-fast-whisper#115]
I tried using a basic pipeline with a fine-tuned hindi whisper-large-v2 model , the gives decent accuracy but adds words to the end of the sentence. I'm aware that hallucinations is a known issue , couldn't find a decent fix for it though.

Code:
```
import torch
from transformers import pipeline

pipe = pipeline(
    "automatic-speech-recognition",
    model='/mnt/batch/tasks/shared/LS_root/mounts/clusters/tejasw1/code/Users/Tejasw/hindi_models/whisper-large-hi-noldcil',
    torch_dtype=torch.float16,
    device="cuda", # or mps for Mac devices
    model_kwargs={"use_flash_attention_2": True}, # set to False for old GPUs
)

# pipe.model = pipe.model.to_bettertransformer() # only if `use_flash_attention_2` is set to False

outputs = pipe("test_audio.wav",
               chunk_length_s=30,
               batch_size=24,
            #    return_timestamps=True,
               )

outputs
```

Transcribed sentence:
```
प्रस्तुत मामले में वादी मुकदमा शिवेंद्र द्वारा दिनांक 1872019 को थारासोरो पर लिखित तहरीर इस आशय के प्रस्तुत की गई कि प्रार्थी ने अपनी पुत्री की शादी लगभग दस साल पहले सतीश पुत्र दुलार सिंह निवासी दतलाना थाने सोरो के साथ की थी और अपने अपनी हैसियत के मुताबिक लगभग पांच लाख रुपये खर्च किए थे लेकिन सुमन की सास तथा जेठ गिरपाल सिंह व भपेंद्र सिंह तथा सतीश व मुलायम सिंह पुत्र सुधीर सिंह दिए गए दान दहेज से संतुष्ट नहीं थे और उसकी लड़की से अतिरिक्त दहेज में मोटर साइकिल व सोने की जंजीर और अगूठी की मांग करते थे और उसके लिए यह बहुत अधिक विश्वास नहीं था
```

Should be:
```
प्रस्तुत मामले में वादी मुकदमा शिवेंद्र द्वारा दिनांक 1872019 को थारासोरो पर लिखित तहरीर इस आशय के प्रस्तुत की गई कि प्रार्थी ने अपनी पुत्री की शादी लगभग दस साल पहले सतीश पुत्र दुलार सिंह निवासी दतलाना थाने सोरो के साथ की थी और अपने अपनी हैसियत के मुताबिक लगभग पांच लाख रुपये खर्च किए थे लेकिन सुमन की सास तथा जेठ गिरपाल सिंह व भपेंद्र सिंह तथा सतीश व मुलायम सिंह पुत्र सुधीर सिंह दिए गए दान दहेज से संतुष्ट नहीं थे और उसकी लड़की से अतिरिक्त दहेज में मोटर साइकिल व सोने की जंजीर और अगूठी की मांग करते थे |
```

**Comment by sanchit-gandhi** [comment]
Since we fine-tune Whisper on a different distribution of audio data to that used for pre-training, we shift the distribution of the weights to favour a new set of hyper-parameters. In particular, it's highly likely the audios you used during fine-tuning were on average less than 30s, whereas Whisper was pre-trained on 30s audios. To compensate, you can try setting a lower value of `chunk_length_s`. In fact, the best thing to do is 'tune' `chunk_length_s` by running evaluation several times on your audio:

```python
import torch
from transformers import pipeline
from transformers.utils import is_flash_attn_2_available

pipe = pipeline(
    "automatic-speech-recognition",
    model='/mnt/batch/tasks/shared/LS_root/mounts/clusters/tejasw1/code/Users/Tejasw/hindi_models/whisper-large-hi-noldcil',
    torch_dtype=torch.float16,
    device="cuda", # or mps for Mac devices
    model_kwargs={"use_flash_attention_2": is_flash_attn_2_available()},
    batch_size=24,
)

if not is_flash_attn_2_available():
    pipe.model = pipe.model.to_bettertransformer()

for chunk_length_s in [10, 15, 20, 25, 30]:
    outputs = pipe("test_audio.wav", chunk_length_s=chunk_length_s)
    print(f"chunk_length_s: {chunk_length_s}, text: {outputs['text']}")
```
(Also see here the automatic selection of FA2 vs Better Transformer -> this is a nice way of automatically configuring them @Vaibhavs10)
